src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SnackBar.java

```diff
@@ -16,58 +16,24 @@
 
 package io.cdap.cdap.ui.stepsdesign;
 
-import io.cdap.cdap.ui.utils.Constants;
 import io.cdap.cdap.ui.utils.Helper;
 import io.cdap.e2e.utils.ElementHelper;
 import io.cdap.e2e.utils.SeleniumDriver;
 import io.cdap.e2e.utils.WaitHelper;
-import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import org.junit.Assert;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
 
-import java.util.ArrayList;
-import java.util.List;
 
 public class SnackBar {
-  @Given("Navigate to Home Page")
-  public void navigateToHomePage() {
-    try {
-      SeleniumDriver.openPage(Constants.WRANGLE_HOME_URL);
-      WaitHelper.waitForPageToLoad();
-    } catch (Exception e) {
-      System.err.println("error:" + e);
-    }
-  }
-
-  @Then("Click on the data exploration card")
-  public void clickOnTheDataExplorationCard() {
-    try {
-      WaitHelper.waitForPageToLoad();
-      List<String> productName = new ArrayList<String>();
-      List<WebElement> allProductsName = SeleniumDriver.getDriver().findElements(
-          By.xpath(".//*[@data-testid='wrangler-home-ongoing-data-exploration-card']"));
-      ElementHelper.clickOnElement(allProductsName.get(0));
-      String url = SeleniumDriver.getDriver().getCurrentUrl();
-      Assert.assertTrue(url.contains("http://localhost:11011/cdap/ns/default/wrangler-grid"));
-    } catch (Exception e) {
-      System.err.println("error:" + e);
-    }
-  }
-
   @Then("Click on the Snackbar close icon")
   public void verifyTheSnackbarPopUpIsComingOrNot() {
-    try {
-      WaitHelper.waitForPageToLoad();
-      WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("snackbar-alert"));
-      Assert.assertTrue(ElementHelper.isElementDisplayed(Helper.locateElementByTestId("snackbar-alert")));
-      WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("snackbar-close-icon"));
-      WaitHelper.waitForElementToBeClickable(Helper.locateElementByTestId("snackbar-close-icon"));
-      WebElement ele = SeleniumDriver.getDriver().findElement(By.xpath("//*[@data-testid = 'snackbar-close-icon']"));
-      ElementHelper.clickOnElement(ele);
-    } catch (Exception e) {
-      System.err.println("error:" + e);
-    }
+    WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("snackbar-alert"));
+    Assert.assertTrue(ElementHelper.isElementDisplayed(Helper.locateElementByTestId("snackbar-alert")));
```

Comment:
this is redundant when you have the line above, because if the element is not displayed, line 32 fails and you'll never get here. You can remove line 32, they're interchangeable.

Reply:
@sudhadivami bump

Reply:
@mrahanjam @sechegaray 
1. we need this waitForElementToBeDisplayed to capture both the page load as well as the snack bar element to be displayed
2. this way we can avoid using waitForPageLoad and then another way to delay the snack bar element to load

3. and in this case, we have to wait for the element to load as the page and element are loading at the same time and we need to check this element to write the e2e tests for the snack bar element